zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/GroupQueryControllerTest.java

```diff
@@ -95,4 +100,111 @@ void shouldReturnNotFoundOnGetNonExistingGroup() {
     // then
     verify(groupServices, times(1)).getGroup(groupKey);
   }
+
+  @Test
+  void shouldSearchGroupsWithEmptyQuery() {
+    // given
+    final var groupKey1 = 111L;
+    final var groupKey2 = 222L;
+    final var groupKey3 = 333L;
+    final var groupName1 = "Group 1";
+    final var groupName2 = "Group 2";
+    final var groupName3 = "Group 3";
+    when(groupServices.search(any(GroupQuery.class)))
+        .thenReturn(
+            new SearchQueryResult.Builder<GroupEntity>()
+                .total(3)
+                .sortValues(new Object[] {})
+                .items(
+                    List.of(
+                        new GroupEntity(groupKey1, groupName1, Set.of()),
+                        new GroupEntity(groupKey2, groupName2, Set.of()),
+                        new GroupEntity(groupKey3, groupName3, Set.of())))
+                .build());
+
+    // when / then
+    webClient
+        .post()
+        .uri("%s/search".formatted(GROUP_BASE_URL))
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue("{}")
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(
+            """
+          {
+             "items": [
+               {
+                 "key": %d,
+                 "name": "%s"
+               },
+               {
+                 "key": %d,
+                 "name": "%s"
+               },
+               {
+                 "key": %d,
+                 "name": "%s"
+               }
+             ],
+             "page": {
+               "totalItems": 3,
+               "firstSortValues": [],
+               "lastSortValues": []
+             }
+           }"""
+                .formatted(groupKey1, groupName1, groupKey2, groupName2, groupKey3, groupName3));
+
+    verify(groupServices).search(new GroupQuery.Builder().build());
+  }
+
+  @Test
+  void shouldSortAndPaginateSearchResult() {
```

Comment:
❌ I'm missing test cases for the exceptional cases. What if the field in sort is invalid? What if the provided order is invald? What if I provide invalid values to page, such as a negateive limit?

Reply:
Good point, I've added them.